### PR TITLE
General data

### PR DESCRIPTION
**Pull request by natgeo89**
1. Create new mock files. Now, **one source API** gives info about covid info, population and flag. So there are no more problems with country naming. Use mocks in './src/js/mocks/' for testin your module.
2. File getApiData.js in './src/js/' includes 2 functions, that request need data, processes this data and return result in convenient form. Remember, that these functions returns Promise.
3. Fix linter errors in webpack.config, so conflict in merging should be. Choose right variant)

### CODE REVIEW

File: covid-dashboard/src/js/getApiData.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+// for table, list with flags
+async function getCountriesAndGlobalInfo() {
+  const urlCountries = 'https://disease.sh/v3/covid-19/countries';
```
Comment (IvanPresmytsky) [inline covid-dashboard/src/js/getApiData.js]: Could you please move the common part of url in separate variable and reuse between functions?

File: covid-dashboard/src/js/getApiData.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    todayCases,
+    deaths,
+    todayDeaths,
+    recovered,
+    todayRecovered,
+    population,
+    updated,
+  };
+  console.log(JSON.stringify(countriesInfo, null, 2)); // just for control
+  return countriesInfo;
```
Comment (IvanPresmytsky) [inline covid-dashboard/src/js/getApiData.js]: It's better to split such a big function to smaller ones.
fetchContriesData, fetchGlobalData, getContriesData, get Global Data, getContriesAndGlobalData.
The first two just fetch data from different endpoints, the second two map to appropriate data format, and the last one aggregates both.

File: covid-dashboard/src/js/getApiData.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}
+getCountriesAndGlobalInfo();
```
Comment (IvanPresmytsky) [inline covid-dashboard/src/js/getApiData.js]: Why do you need to call it here?

File: covid-dashboard/src/js/getApiData.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    console.log(JSON.stringify(countryStats, null, 2));// just for control
+    return countryStats;
+  }
+  const url = `https://disease.sh/v3/covid-19/historical/${country}?lastdays=1000`;
+  const response = await fetch(url);
+  const countryStats = await response.json();
+  const { cases, deaths, recovered } = countryStats.timeline;
+  console.log(JSON.stringify({ cases, deaths, recovered }, null, 2)); // just for control
+  return { cases, deaths, recovered };
+}
```
Comment (IvanPresmytsky) [inline covid-dashboard/src/js/getApiData.js:120]: The same here: 2 fetch function and 1 function-aggregator

**Review by IvanPresmytsky — COMMENTED** [review]
Great job! Please check the comments. 
They related to single responsibility principle.

File: covid-dashboard/src/js/getApiData.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  return fetch(url);
+}
+
+async function fetchHistoricGlobalInfo() {
+  const url = `${urlCovidBase}/historical/all?lastdays=1000`;
+  return fetch(url);
+}
+
+async function getHistoricCountryInfo(country) {
+  const response = await fetchHistoricCountryInfo(country).then((result) => result.json());
```
Comment (IvanPresmytsky) [inline covid-dashboard/src/js/getApiData.js]: You can move toJson logic to fetch functions to avoid duplication

**Review by IvanPresmytsky — APPROVED** [review]

**Review by confesssa — APPROVED** [review]

**Pull request merged**